### ISSUE TITLE
Deprecations in Ruby 2.7 

### ISSUE THREAD

**Issue by michaelmcchen** [DataDog/dd-trace-rb#1542]
dd-trace version v0.49.0
Rails version 6.0.3.6
Ruby version 2.7.1

When running any sort of Rails command that causes Rails to auto-load, such as `rails console`, the following deprecation warnings appear. This seems like something that could be easily fixed to improve general developer experience :).

```
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:80: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:84: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:88: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:113: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:117: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:121: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:216: warning: The called method `initialize' is defined here
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:62: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/ddtrace-0.49.0/lib/ddtrace/configuration/agent_settings_resolver.rb:29: warning: The called method `initialize' is defined here
```

**Comment by ivoanjo** [comment]
Hey @michaelmcchen, thanks for the report, and sorry for the added noise in your dev workflow!

Those warnings are actually a Ruby bug, affecting Ruby versions 2.7.0 and 2.7.1. The 2.7.2 release notes don't explicitly mention it (so I can't link directly to the fix), but they should be gone on 2.7.2+.

Would moving to 2.7.3 be an option for you?

**Comment by michaelmcchen** [comment]
Thanks so much for the quick response @ivoanjo! Running the app locally on 2.7.3 did fix the issue 😄. I'll look into formally upgrading on our end.
